### PR TITLE
Added bios resource provider

### PR DESCRIPTION
**Pull request by anupamaloke**
##### SUMMARY

Added BIOS resource provider for configuring the BIOS attributes:

- Create / Update - updates the BIOS attributes details. If a TaskMonitor/Job URI is returned in response header, then the ```bios_config_job_uri``` is updated with the TaskMonitor/Job URI. On subsequent apply, evaluate the ```bios_config_job_uri``` first. If the ```bios_config_job_uri``` contains a Job URI, then fetch the task status from the server BMC to determine whether the task is still running or has completed. If it is still running, then don't perform any configuration, else clear the ```bios_config_job_uri``` and patch the new attribute values.
- Destroy - clears the ID

```
provider "redfish" {
  redfish_endpoint = "https://100.71.112.57"
  user = "root"
  password = "calvin"
  ssl_insecure = true
}

resource "redfish_bios" "bios" {
  attributes = {
    "NumLock" = "On"
  }
  settings_apply_time = "OnReset"
}

```
```
$ terraform apply

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # redfish_bios.bios will be created
  + resource "redfish_bios" "bios" {
      + attributes          = {
          + "NumLock" = "On"
        }
      + bios_config_job_uri = (known after apply)
      + id                  = (known after apply)
      + settings_apply_time = "OnReset"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

redfish_bios.bios: Creating...
redfish_bios.bios: Creation complete after 6s [id=/redfish/v1/Systems/System.Embedded.1/Bios]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```
```
$ terraform show
# redfish_bios.bios:
resource "redfish_bios" "bios" {
    attributes          = {
        "AcPwrRcvry"                   = "Last"
        "AesNi"                        = "Enabled"
        "AssetTag"                     = ""
        "AuthorizeDeviceFirmware"      = "Disabled"
        "BootMode"                     = "Bios"
        "BootSeqRetry"                 = "Enabled"
        ...
      <Truncated for brevity>
    }
    bios_config_job_uri = "/redfish/v1/TaskService/Tasks/JID_996862126740"
    id                  = "/redfish/v1/Systems/System.Embedded.1/Bios"
    settings_apply_time = "OnReset"
}
```

Relates OR Closes #0000

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

resource_redfish_bios

##### ADDITIONAL INFORMATION

```paste below

```

#### Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note

FEATURES:
* **New Resource** `resource_redfish_bios`  [GH-3]

```

#### Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

#### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

File: redfish/resource_redfish_bios.go [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+func updateBiosAttributes(d *schema.ResourceData, bios *redfish.Bios, attributes map[string]interface{}) error {
+
+	payload := make(map[string]interface{})
+	payload["Attributes"] = attributes
+
+	if settingsApplyTime, ok := d.GetOk("settings_apply_time"); ok {
+		allowedValues := bios.AllowedAttributeUpdateApplyTimes()
+		allowed := false
+		for i := range allowedValues {
```
Comment (mikeletux) [inline redfish/resource_redfish_bios.go:230]: Hi Anupam,

If we are using *validation.StringInSlice* in **ValidateFunc** in the schema to check if the value given on **settings_apply_time** is within four values, why do we check that again manually?

File: redfish/resource_redfish_bios.go [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+	log.Printf("[DEBUG] Beginning update")
+	var diags diag.Diagnostics
+
+	conn := m.(*gofish.APIClient)
+
+	// check if there is already a bios config job in progress
+	// if yes, then check the current status of the job. If it
+	// has not completed yet, then don't perform another operation
+	pending := false
```
Comment (mikeletux) [inline redfish/resource_redfish_bios.go:65]: Here maybe we could change a little the behavior.
If there is a job scheduled, and you call this method, silently it won't trigger any warning, which may be confusing for the user. We might trigger a warning using **diag.Diagnostics** to let the user know that no operation will be performed until that job is finished.

File: redfish/resource_redfish_bios.go (hunk 1 above)
Comment (anupamaloke) [reply redfish/resource_redfish_bios.go:230]: @mikeletux, the schema validation check for ```settings_apply_time``` checks it against all possible values as per the spec, however, a BMC might only support a subset of those. That is why I added the check to make sure that the value given is being supported by the BMC or not.

File: redfish/resource_redfish_bios.go (hunk 2 above)
Comment (anupamaloke) [reply redfish/resource_redfish_bios.go:65]: @mikeletux, I agree. I will make the changes

**Comment by anupamaloke** [comment]
@mikeletux, I have updated the code in line with the review comments. Please take a look and let me know if any further changes are needed.

**Pull request merged**